lib/build.ts

```diff
@@ -20,13 +20,19 @@ import * as _ from 'lodash';
 import { Builder, BuildHooks, FromTagInfo } from 'resin-docker-build';
 import * as semver from 'semver';
 import type * as Stream from 'stream';
+import { DockerfileParser } from 'dockerfile-ast';
 
 import type { SecretsPopulationMap } from './build-secrets';
 import type { BuildTask } from './build-task';
 import { BuildProcessError } from './errors';
 import { pullExternal } from './external';
 import { LocalImage } from './local-image';
 import type { RegistrySecrets } from './registry-secrets';
+import {
+	DockerImageManifest,
+	RegistryClient,
+	MEDIATYPE_MANIFEST_V1,
+} from './registry/registry-client';
```

Comment:
Similar to my other comment, this one could be split in two:

* If needed to annotate function signatures, an `import type` clause here that is only used by TypeScript for type checking and disappears in the compiled JavaScript code, thus having no runtime implications:
```ts
import type { DockerImageManifest } from './registry/registry-client';
```
* An import clause inside `getRecommendedPlatformhHandling`, closer to where it gets used:
```
const { RegistryClient, MEDIATYPE_MANIFEST_V1 } = await import('./registry/registry-client');
```

Reply:
@pdcastro this then result in the same code getting called each time `getRecommendedPlatformHandling` (once per call of `runBuildTask`.  Seems that, although the point is well taken especially in the previous case, perhaps it might be non-performant here?

Reply:
I think you are right, mainly because of how `balena-builder` uses `balena-multibuild`. It is a trade off between spending import time when `balena-multibuild` is imported for the first time (which contributes to interactive CLI usage "feeling slow"), _vs._ import time when `getRecommendedPlatformHandling` is called (which is worse for long-lived processes like `balena-builder`).  The result of `import('./registry/registry-client')` will be cached, so only the first `import` call will be relatively expensive, but even so it costs something. For `balena-cli`, `getRecommendedPlatformHandling` would be called a small-ish number of time, say 20 times for 20 services (devenv / BoB), and each call would dilute the import time with the image build time for the service that can take minutes. But for `balena-builder`, `getRecommendedPlatformHandling` could be called thousands of times over weeks, months, years...

Reply:
Ah, ok.  Didn't know that about the caching.  Good to know.